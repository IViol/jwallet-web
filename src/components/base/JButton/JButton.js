/* @flow */

import React from 'react'
import classnames from 'classnames'
import { empty, prop } from 'ramda'

import JText from '../JText'
import JIcon from '../JIcon'
import './JButton.scss'

type Props = {
 text?: string,
 color: 'white' | 'blue',
 large?: boolean,
 minimal?: boolean,
 onClick?: Function,
 iconName?: string,
 disabled?: boolean,
 isLoading?: boolean,
}

const JButton = ({
  text,
  color,
  large,
  onClick,
  minimal,
  iconName,
  disabled,
  isLoading,
}: Props) => (
  <div
    onClick={disabled || onClick}
    className={classnames('JButton', {
      '-loading': isLoading,
    }, prop(color, {
      blue: '-blue',
      white: '-white',
    }), large ? '-large' : '-regular', minimal && '-minimal')}
  >
    {iconName && (
      <div className='icon'>
        <JIcon name={iconName} size='small' />
      </div>
    )}
    {text && (
      <div className='text'>
        <JText value={text} />
      </div>
    )}
  </div>
)

JButton.defaultProps = {
  text: undefined,
  large: false,
  onClick: empty,
  minimal: false,
  iconName: undefined,
  disabled: false,
  isLoading: false,
  withBorder: false,
}

export default JButton
