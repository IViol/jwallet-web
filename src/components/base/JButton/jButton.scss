.Button {

  &.-loading {

  }

  &.-bordered {

  }

  &.-disabled {

  }

  &.-blue {

  }

  &.-white {
    
  }

  > .text {

  }

  > .icon {

  }

}
