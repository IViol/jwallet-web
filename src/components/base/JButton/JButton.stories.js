
/* @flow */

import React from 'react'
import { storiesOf } from '@storybook/react'

import JButton from './JButton'

storiesOf('Button', module)
  .add('Big blue', () => (
    <JButton
      text='Button'
      size='big'
      color='blue'
    />
  ))
  .add('Big white', () => (
    <JButton
      text='Button'
      size='big'
      color='white'
    />
  ))
  .add('Small blue', () => (
    <JButton
      text='Button'
      size='small'
      color='blue'
    />
  ))
  .add('Small white', () => (
    <JButton
      text='Button'
      size='small'
      color='white'
    />
  ))
