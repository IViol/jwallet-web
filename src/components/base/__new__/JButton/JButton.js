/* @flow */

import React from 'react'
import { pure } from 'recompose'
import classnames from 'classnames'
import { empty, prop } from 'ramda'

import JText from '../JText'
import JIcon from '../JIcon'
import './JButton.scss'

type Props = {
 text?: string,
 color: 'white' | 'blue',
 large?: boolean,
 minimal?: boolean,
 onClick?: Function,
 iconName?: string,
 iconSize?: 'small' | 'medium',
 disabled?: boolean,
 isLoading?: boolean,
}

const JButton = ({
  text,
  color,
  large,
  onClick,
  minimal,
  iconName,
  iconSize,
  disabled,
  isLoading,
}: Props) => (
  <div
    onClick={disabled || onClick}
    className={
      classnames(
        'JButton', {
          '-loading': isLoading,
          '-with-text': text,
        }, prop(color, {
          blue: '-blue',
          white: '-white',
        }),
        large ? '-large' : '-regular',
        minimal && '-minimal'
      )
    }
  >
    {iconName && (
      <div className='icon'>
        <JIcon name={iconName} size={iconSize} />
      </div>
    )}
    {text && (
      <div className='text'>
        <JText
          value={text}
          variants={[
            minimal
              ? { white: 'white', blue: 'blue' }[color]
              : { white: 'blue', blue: 'white' }[color],
            'bold',
            'normal',
          ]}
        />
      </div>
    )}
  </div>
)

JButton.defaultProps = {
  text: undefined,
  large: false,
  onClick: empty,
  minimal: false,
  iconName: undefined,
  disabled: false,
  iconSize: 'small',
  isLoading: false,
}

export default pure(JButton)
