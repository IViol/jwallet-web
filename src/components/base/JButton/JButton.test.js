/* @flow */

import React from 'react'
import renderer from 'react-test-renderer'

import JButton from './JButton'

describe('COMPONENT: JButton', () => {
  test('Blue', () => {
    expect(renderer.create(
      <JButton
        size='big'
        color='blue'
      />
    ).toJSON()).toMatchSnapshot()
  })
})
