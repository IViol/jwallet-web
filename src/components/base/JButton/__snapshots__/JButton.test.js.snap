// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: JButton Blue 1`] = `
<div
  className="JButton -blue -regular"
  onClick={[Function]}
/>
`;
